provider/datagen/src/transform/segmenter/dictionary.rs

```diff
@@ -50,14 +47,13 @@ macro_rules! implement {
             }
         }
 
-        impl IterableDataProviderCached<$marker> for DatagenProvider {
-            fn supported_locales_cached(
+        impl IterableDataProvider<$marker> for DatagenProvider {
+            fn supported_requests(
```

Comment:
Question: Why are you changing this from `IterableDataProviderCached` to `IterableDataProvider`; that seems regressive

Reply:
There seems to be a bug in the caching which I need to chase down, it throws missing locale errors.

Reply:
couldn't reproduce, maybe it was something else